src/model/test/Surface_GTest.cpp

```diff
@@ -3798,3 +3800,40 @@ TEST_F(ModelFixture, Issue_4374) {
     EXPECT_TRUE(surface.subSurfaces().size() <= 1);
   }
 }
+
+TEST_F(ModelFixture, 4678_SurfaceGlassUFactorSqlError) {
+  // Test for #4678 - Glass U Factor sql error
+
+  // This one has fenestration that includes WindowProperty:FrameAndDivider
+  openstudio::path idfPath = resourcesPath() / toPath("energyplus/FrameAndDivider/in.idf");
+  energyplus::ReverseTranslator reverseTranslator;
+  ASSERT_NO_THROW(reverseTranslator.loadModel(idfPath));
+  OptionalModel _model = reverseTranslator.loadModel(idfPath);
+  ASSERT_TRUE(_model);
+  Model model = _model.get();
+
+  openstudio::path sqlPath = resourcesPath() / toPath("energyplus/FrameAndDivider/eplusout.sql");
+  openstudio::SqlFile sqlFile = openstudio::SqlFile(sqlPath);
+  ASSERT_TRUE(sqlFile.connectionOpen());
+
+  model.setSqlFile(sqlFile);
+  ASSERT_TRUE(model.sqlFile());
+
+  OptionalSurface surface = model.getModelObjectByName<Surface>("Story 1 Core Space Exterior Wall");
+  ASSERT_TRUE(surface);
+
+  OptionalConstructionBase oConstruction = surface->construction();
+
+  ASSERT_TRUE(oConstruction);
+
+  EXPECT_TRUE(oConstruction->isOpaque());
+  EXPECT_FALSE(oConstruction->isFenestration());
+
+  ASSERT_TRUE(surface->uFactor());
+  double uFactor = surface->uFactor().get();
+  EXPECT_TRUE(openstudio::equal(0.310, uFactor, 1.0E-3));
```

Comment:
`EXPECT_DOUBLE_EQ(val1, val2)` is meant for this, and will report a nicer output in case it fails. 

> Verifies that the two double values val1 and val2 are approximately equal, to within 4 ULPs from each other.

If you need a specific tolerance (I don't think you do), then use `EXPECT_NEAR(val1, val2, abs_error)`